File: Elixir/fib/fib.c
#include <stdio.h>
#include <stdlib.h> 
int fib(int n) {
    if (n == 1 || n == 0) {
        return n;
    }
    return fib(n-1) + fib(n-2);
}

int main(int argc, char **argv) {
    // printf("%d\n", argc);
    if (argc >= 2) {
        char* buf = argv[1];
        int val = atoi(buf);
        // printf("%d\n", val);
        int ans = fib(val);
        printf("%d\n", ans);
    } else {
        printf("could not read command line arguments");
    }
}

File: Elixir/fib/fib.ex
defmodule Maths do
  def fib(n) do
    case n do
      0 -> 0
      1 -> 1
      _ -> fib(n - 1) + fib(n - 2)
    end
  end
end

# Maths.fib(20)
# |> IO.inspect()

if System.argv() |> length() >= 1 do
  System.argv()
  |> Enum.at(0)
  |> Integer.parse()
  |> case do
    :error -> IO.puts("Error in reading command line args")
    {x, _} -> Maths.fib(x) |> IO.inspect()
  end
else
  IO.puts("Failed to read command line args: [#{System.argv()}]")
end
